### DISCUSSION TITLE
Routing to entrance=main

### DISCUSSION THREAD

**Discussion by Lorenzopesci** [osmandapp/OsmAnd#19287] [Maps Related]
Some questions:
1 routing
I have problems with routing for large buildings on Osmand.
If I specify a large building or place as the destination, the arrival point is set to the centre of the area.
This also happens when there is entrance=main on the perimeter, but the point is not considered by Osmand as an entrance to the building.
If there are several roads around the area, Osmand chooses the one closest to the building, but not the one near the entrance, so I could arrive at the back by mistake.
Normally on Open street map it is correct to enter the name on the area and not on the point.
How can the result of finding the correct destination be achieved?

I ask the APP, for example, to reach "Castel Sant'Angelo" in Rome and I expect the APP to take me to the front of the entrance using the data available to it (entrance=main https://www.openstreetmap.org/node/5730657087).
Instead it takes me to the back, ignoring the tags specifically placed on the perimeter of the building.

2 Holy Wayside Shrines
In hiking, holy shrines are important landmarks, but on Osmand in Offroad mode (the clearest for hiking), the shrines are not visible.

3 Monumental trees
Monumental trees are not characterised, although they are of great interest: they are only seen if they have a name, but not all of them have a name, even if they have denotation=natural_monument

Lorenzo

<table style="width:100%">
  <tr>
<th><img src="https://github.com/osmandapp/OsmAnd/assets/65073339/b793670f-dda4-4ca3-a522-382c4446fa7b"  width="300" height="auto"/> </td>
  </tr>
</table>

**Comment by stalker314314** [comment]
This seems similar to https://github.com/osmandapp/OsmAnd/issues/7520 or https://github.com/osmandapp/OsmAnd/issues/13431?
